### PR TITLE
feat(Mandatory Modules): Add Mandatory optional field to the ModuleTemplate Spec

### LINKED ISSUE #1050: [Mandatory Modules] - ModuleTemplate spec
### Description

As part of the new required epic of Mandatory Modules, we have need to enhance our existing Custom Resource Definition (CRD) for ModuleTemplate. The goal of this issue is to introduce a new optional specification called 'mandatory' into the ModuleTemplate CRD Spec.
The new attribute would be a boolean field (true or false) and allow module teams to define whether a specific module is mandatory or not, meaning it will be installed to the remote cluster without the knowledge of the user.

### Reasons

_No response_

### Acceptance Criteria

- [x] Successfully introduce and implement the new optional field `spec.mandatory` in ModuleTemplate CRD. The new field should be of type boolean.
- [x] The field should be optional
- [x] After the API upgrade it should be mandatory on the CR level (create a follow-up issue, this should be filled by the CLI)
- [x] API docs of the ModuleTemplate should be updated

### Feature Testing

Unit tests

### Testing approach

_No response_

### Attachments

_No response_

### PR DESCRIPTION
**Pull request by nesmabadr**
**Description**

Changes proposed in this pull request:

- Add `mandatory` optional boolean field to the ModuleTemplate spec
- Update API documentation for the ModuleTemplate to add the mandatory field documentation

**Related issue(s)**
Resolves #1050

### CODE REVIEW

**Review by jeremyharisch — COMMENTED** [review]
Looks good, but documentation is missing.

Please write a small description about the (planed) usage of this mandatory field:  https://github.com/kyma-project/lifecycle-manager/blob/main/docs/technical-reference/api/moduleTemplate-cr.md

Kind of the same as you wrote in the description in the CRD definition, but with a bit more details and more easy to read,

File: docs/technical-reference/api/moduleTemplate-cr.md [hunk 1]
```diff
@@ -87,6 +87,11 @@ The core of any ModuleTemplate CR, the descriptor can be one of the schemas ment
 
 By default, it will most likely be easiest to use [Kyma CLI](https://github.com/kyma-project/cli/tree/main) and its `create module` command to create a template with a valid descriptor, but it can also be generated manually, for example using [OCM CLI](https://github.com/open-component-model/ocm/tree/main/cmds/ocm).
 
+### `.spec.mandatory`
+
+The `mandatory` field indicates whether the module should be installed to all runtime clusters without any interaction from the user.
+Mandatory modules do not appear in the Kyma CR of the runtime cluster, and they have the same configuration across all runtime clusters.
```
Comment (jeremyharisch) [inline docs/technical-reference/api/moduleTemplate-cr.md:93]:
```suggestion
Mandatory modules do not appear in the Kyma CR `.status` and `.spec.modules`, furthermore they have the same configuration across all runtime clusters.
```

**Review by grego952 — APPROVED** [review]

File: docs/technical-reference/api/moduleTemplate-cr.md [hunk 2]
```diff
@@ -87,6 +87,11 @@ The core of any ModuleTemplate CR, the descriptor can be one of the schemas ment
 
 By default, it will most likely be easiest to use [Kyma CLI](https://github.com/kyma-project/cli/tree/main) and its `create module` command to create a template with a valid descriptor, but it can also be generated manually, for example using [OCM CLI](https://github.com/open-component-model/ocm/tree/main/cmds/ocm).
 
+### `.spec.mandatory`
+
+The `mandatory` field indicates whether the module is installed in all runtime clusters without any interaction from the user.
```
Comment (jeremyharisch) [inline docs/technical-reference/api/moduleTemplate-cr.md:92]:
```suggestion
The `mandatory` field indicates whether the module is installed in all runtime clusters without any interaction from the user.
```

That is wrong, it is not about that it `is` installed, but it `should be` installed. I think `is` is very confusing here. That mandatory field does not indicate the current state, but how the module template should be treated as

**Review by jeremyharisch — APPROVED** [review]

File: api/v1beta2/moduletemplate_types.go [hunk 3]
```diff
@@ -80,6 +80,11 @@ type ModuleTemplateSpec struct {
 	// +kubebuilder:validation:MinLength:=3
 	Channel string `json:"channel"`
 
+	// Mandatory indicates whether the module is mandatory. It is used to enforce the installation of the module with
+	// its configuration in all runtime clusters.
+	// +optional
+	Mandatory bool `json:"mandatory"`
```
Comment (ruanxin) [inline api/v1beta2/moduletemplate_types.go:86]: This field should also introduce into `v1beta1`, `v1beta1` is current storage version, if you only introduce here, the value will be lost during conversion. @nesmabadr  @jeremyharisch

**Pull request merged**